cluster-autoscaler/cloudprovider/gce/cache.go

```diff
@@ -184,6 +212,7 @@ func (gc *GceCache) SetMigInstances(migRef GceRef, instances []cloudprovider.Ins
 
 	gc.removeMigInstances(migRef)
 	gc.instances[migRef] = append([]cloudprovider.Instance{}, instances...)
+	gc.instancesRefreshTime[migRef] = gc.timeProvider.Now()
```

Comment:
since we're not using `gc.instancesRefreshTime` internally in GceCache, I don't think it makes sense to set the state in the package either. Especially since the refresh is happening externally and we're just setting the state of the cache.

I would look to create a meta func along the lines of `SetMigInstances(migRef GceRef, instances []cloudprovider.Instance, updateTimestamp time.Time)`, signifying when this information was generated.

Thoughts?

p.s. likewise we shouldn't have a need for `timeProvider` since it's the callers responsibility to provide that.

Reply:
That's a good point, updated the `SetMigInstances`